flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStore.java

```diff
@@ -233,68 +229,90 @@ public int getNumberOfRetainedCheckpoints() {
 	}
 
 	@Override
-	public void shutdown() throws Exception {
-		LOG.info("Shutting down");
+	public void shutdown(JobStatus jobStatus) throws Exception {
```

Comment:
Our only option would be to wrap in our own Exception, because Curator is throwing the general `Exception`.